src/lib/pages/backoffice/Document/DocumentDetails/DocumentPurchaseOrders/DocumentPurchaseOrders.js

```diff
@@ -0,0 +1,148 @@
+
+import { dateFormatter } from '@api/date';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withCancel } from '@api/utils';
+import { SeeAllButton } from '@components/backoffice/buttons/SeeAllButton';
+import { orderApi } from '@api/acquisition';
+import { ResultsTable } from '@components/ResultsTable/ResultsTable';
+import { Loader } from '@components/Loader';
+import { Link } from 'react-router-dom';
+import { AcquisitionRoutes } from '@routes/urls';
+import { Message } from 'semantic-ui-react';
+
+export default class DocumentPurchaseOrders extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      orders: {},
+      isLoading: true,
+      error: false
+    };
+  }
+
+  componentDidMount() {
+    const {
+      documentDetails: { pid },
+    } = this.props;
+    this.fetchPurchaseOrders(pid)
+  }
+
+  componentWillUnmount() {
+    this.cancellablePurchaseOrders && this.cancellablePurchaseOrders.cancel();
+  }
+
+
+  seeAllButton = () => {
+    const {
+      documentDetails: { pid },
+    } = this.props;
+    const path = AcquisitionRoutes.ordersListWithQuery(
+      orderApi
+        .query()
+        .withDocPid(pid)
+        .qs()
+    );
+    return <SeeAllButton to={path} />;
+  };
+
+  fetchPurchaseOrders = async (pid) => {
+    try {
+      this.cancellablePurchaseOrders = withCancel(
+        orderApi.list(orderApi.query().withDocPid(pid).qs())
+      );
+      const response = await this.cancellablePurchaseOrders.promise;
+      this.setState({ orders: response.data, error: false, isLoading: false});
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({error: true, isLoading: false});
+      }
+    }
+  };
+
+  viewDetails = ({ row }) => {
+    return (
+      <Link
+        to={AcquisitionRoutes.orderDetailsFor(row.id)}
+      >
+        {row.id}
+      </Link>
+    );
+  };
+
+
+  renderTable(data){
+    const { showMaxPurchaseOrders } = this.props;
+    const columns = [
+      { title: 'Order', formatter: this.viewDetails },
+      {
+        title: 'Status',
+        field: 'metadata.status',
+      },
+      {
+        title: 'Provider',
+        field: 'metadata.provider.name',
+      },
+      {
+        title: 'Created date',
+        field: 'created',
+        formatter: dateFormatter,
+      },
+      {
+        title: 'Expected delivery date',
+        field: 'metadata.expected_delivery_date',
+        formatter: dateFormatter,
+      },
+    ];
+
+    return (
+      <ResultsTable
+        data={data.hits}
+        columns={columns}
+        totalHitsCount={data.total}
+        name="Orders"
+        seeAllComponent={this.seeAllButton()}
+        showMaxRows={showMaxPurchaseOrders}
+      />
+    );
+  }
+
+  renderResultsOrEmpty(data, error) {
+    return data.hasOwnProperty("hits")
```

Comment:
I see you are learning from the training! :)
This is very right, however in general we simply use `hits in data` or the **lodash** `_get` function give that we don't mess with the object prototype and to keep it more readable.